src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/tests/latest/test_resource.py

```diff
@@ -881,5 +881,17 @@ def test_invoke_action(self, resource_group):
         self.cmd('resource invoke-action --action capture --ids {vm_id} --request-body {request_body}')
 
 
+class GlobalIdsScenarioTest(ScenarioTest):
+    @ResourceGroupPreparer(name_prefix='cli_test_global_ids')
+    def test_global_ids(self, resource_group):
+
+        self.kwargs.update({
+            'vnet': 'vnet1'
+        })
+        self.kwargs['vnet_id'] = self.cmd('network vnet create -g {rg} -n {vnet}').get_output_in_json()['newVNet']['id']
+        # command will fail if the other parameters were actually used
+        self.cmd('network vnet show --subscription fakesub --resource-group fakerg -n fakevnet --ids {vnet_id}')
```

Comment:
Do we want to validate and return a warning if we see a user specifying parameters that are ultimately ignored?
Less critical for `subscription` but for `-n` and `-g`, a user may have meant to for some reason show that particular resource.